### PR TITLE
fix(api): eliminate false positives in the case-law AST validator

### PR DESCRIPTION
**Pull request by jan-kubica**
The AST validator's reference-text extraction uses cheerio's `.text()`, which renders `<br>` as nothing. Adjacent words separated only by a line break get glued into phantom tokens the AST rightly does not contain (`wraz<br/>z` → `wrazz`), and the validator then reports them as missing words. Polish typography wraps single-letter prepositions (z, w, i, a, o) to the next line with `<br/>`, so pl-courts documents hit the `maxMissingWords` threshold on perfectly parsed content; sampling the newest 447 SAOS judgments through the parser reproduced 3 such false `ast_validation_failed` reports, every phantom token mapping to a `<br/>` in the source. With the fix the same sample validates clean.

Changes:
- Replace `<br>` with a space before extracting reference text, so line breaks are word boundaries on both sides of the comparison.
- Extend the word-extraction letter set beyond Czech (Polish, Slovak, German) so edge diacritics survive `trimNonLetters` instead of silently shrinking the compared word sets.
- Compare inline text and `plainText` whitespace-collapsed in the `INLINE_PLAINTEXT_MISMATCH` check; parsers normalize `plainText` while inlines keep source spacing, and that alone is not a mismatch.
- Log full issue messages and up to 25 missing words on validation failure (court decisions are public documents), so a failure is diagnosable straight from the log line instead of requiring a re-fetch and re-parse.

Regression tests cover the `<br>` word-boundary case, Polish edge diacritics, and whitespace-only inline/plainText differences.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved case-law text validation for line breaks and whitespace differences.
  * Corrected word extraction to preserve Czech, Polish, Slovak and German diacritics.
  * Reduced false missing-word and inline text mismatch reports.
  * Expanded validation diagnostics with more detailed issue information.

* **Chores**
  * Updated automated workflows to preserve environment template files during checks and tests.

### CODE REVIEW

File: apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts [hunk 1]
```diff
@@ -58,7 +58,12 @@ const SKIP_WORDS = new Set([
   "pokračování",
 ]);
 
-const LETTERS = new Set("abcdefghijklmnopqrstuvwxyzáčďéěíňóřšťúůýž");
+// Letters across the supported corpora: base Latin, Czech, Polish,
+// Slovak, German. A letter missing here gets trimmed from word edges,
+// silently shrinking the compared word sets for that language.
+const LETTERS = new Set(
+  "abcdefghijklmnopqrstuvwxyzáäąčćďéěęíĺľňńóôöřŕšśťúůüýžźżß",
+);
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts:66]: Good catch — accepted and implemented in 71796a1214, including regression expectations for edge-ł words (mógł, Łódź).

File: apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts [hunk 2]
```diff
@@ -93,22 +98,25 @@ const extractWords = (text: string): Set<string> => {
...
+      text += " ";
     } else {
       // bold | italic | link — all carry children
-      len += inlineTextLength(node.children);
+      text += inlineText(node.children);
     }
   }
-  return len;
+  return text;
 };
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts:115]: Pushing back. `Inline` is a discriminated union and the else branch is already narrowed to the child-bearing variants (bold | italic | link), so the compiler makes the missing-children case unrepresentable; a new inline variant without children would fail typecheck here rather than crash at runtime. Our conventions prefer exhaustive discriminated-union narrowing over `"key" in obj` probes, and silently skipping an unknown node would hide exactly the content loss this validator exists to catch.

File: apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts [hunk 3]
```diff
@@ -58,7 +58,12 @@ const SKIP_WORDS = new Set([
   "pokračování",
 ]);
 
-const LETTERS = new Set("abcdefghijklmnopqrstuvwxyzáčďéěíňóřšťúůýž");
+// Letters across the supported corpora: base Latin, Czech, Polish,
+// Slovak, German. A letter missing here gets trimmed from word edges,
+// silently shrinking the compared word sets for that language.
+const LETTERS = new Set(
+  "abcdefghijklmnopqrstuvwxyzáäąčćďéěęíĺľňńóôöřŕšśťúůüýžźżß",
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts]: Accepted — same defect as the sibling thread; fixed in 71796a1214 with regression expectations for edge-ł words.

File: apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts [hunk 4]
```diff
@@ -347,19 +363,24 @@ export const validateAndLog = (
...
+  // failure is actionable straight from the log line, without
+  // re-fetching and re-parsing the source.
   if (!result.ok) {
     logger.error("case_law.ingestion.ast_validation_failed", {
       parser: parserName,
       caseNumber,
-      issues: result.issues.length,
-      missingWords: result.stats.missingWords.length,
+      issues: result.issues.map((issue) => `${issue.code}: ${issue.message}`),
+      missingWords: result.stats.missingWords.slice(0, 25),
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/handlers/case-law/ingestion/parsers/validate-ast.ts]: Pushing back. This validator is exercised only by the case-law ingestion parsers, whose sources are published court decisions: public records, anonymized by the issuing courts before publication (the parsers even preserve the anonymization markers). That is why the log deliberately includes the offending tokens — count-only logs made these failures undiagnosable without re-fetching and re-parsing the document, and the words themselves are the diagnostic. No personal or tenant data flows through this path.
